packages/devtools/lib/src/debugger/debugger.dart

```diff
@@ -1161,6 +1242,87 @@ class BreakpointsView implements CoreElementView {
   }
 }
 
+class PopupView extends CoreElement {
+  PopupView(this._scriptsView, this._sourceArea, this._sourcePathDiv,
+      this._popupTextfield)
+      : super('div', classes: 'open-popup');
+
+  CoreElement _sourceArea; // Top div area container of _sourcePathDiv
+  CoreElement _sourcePathDiv; // Text name of the source
+  CoreElement _popupTextfield; // Textfield to show while popup is active.
+
+  ScriptsView _scriptsView;
+
+  String _oldSourceNameTextColor;
+
+  bool _poppedUp = false;
+  bool get isPoppedUp => _poppedUp;
+
+  void showPopup() {
+    _poppedUp = true;
+
+    add(_scriptsView);
+
+    final int sourceAreaWidth = _sourceArea.element.clientWidth;
+
+    _scriptsView.element.element.style.height =
+        '${element.getBoundingClientRect().height - 2}px';
+    _scriptsView.element.element.style.width = '${sourceAreaWidth / 2}px';
+
+    final html.Rectangle r = _sourceArea.element.getBoundingClientRect();
+
+    final int nameHeight = _sourcePathDiv.element.clientHeight;
+
+    int leftPosition = 20; // Default left position.
+    final List<html.Node> elems =
+        html.document.getElementsByClassName('CodeMirror-gutters');
+    if (elems.length == 2) {
+      final html.Element firstGutter = elems[0].firstChild as html.Element;
+      if (firstGutter.style.display != 'none')
+        // Compute total gutter width (parent has both BP and line # gutters).
+        leftPosition = firstGutter.parent.getBoundingClientRect().width + 5;
+    }
+
+    // Set the sourcePathDiv text color to the background color (hide filename).
+    final String bgColor =
+        _sourcePathDiv.element.getComputedStyle().backgroundColor;
+    _oldSourceNameTextColor = _sourcePathDiv.element.getComputedStyle().color;
+    _sourcePathDiv.element.style.color = bgColor;
+
+    _popupTextfield.element.style.height = '${nameHeight}px';
```

Comment:
nit: cascades could be nice here as you are writing 
_popupTextfield.element.style. a lot.

Reply:
There's a bunch of lines that should be cascaded, as I was tweaking everything got a bit out of hand.  Good suggestion. Done.